### ISSUE TITLE
Model schema is invalid: invalid components: schema "Output": extra sibling fields: [title]

### ISSUE THREAD

**Issue by stri8ed** [replicate/cog#1275]
I am attempting to push a new model to Replicate, but it's failing during the schema validation stage.

```
Validating model schema...
ⅹ Model schema is invalid: invalid components: schema "Output": extra sibling fields: [title]
```

The output model used is:

```
class ModelOutput(BaseModel):
    preview: str
    srt_file: Path
    vtt_file: Path
```
Looking at the [cog source](https://github.com/replicate/cog/blob/a6a26b7f7a26bec45c92a8644fed3c454099dd39/python/cog/command/ast_openapi_schema.py#L355), it seems it generates a schema based on the model output, and each field includes a title field:

`"properties":{"preview":{"title":"Preview","type":"string"},"srt_file": etc..`

But the validator is complaining about this field.

What can I do to resolve this?

**Comment by stri8ed** [comment]
> Having the same issue today. An update would be appreciated.

I was able to "solve" it, by downgrading to version 0.8.3

**Comment by xaksis** [comment]
@stri8ed could you state how you downgraded to 0.8.3? I've been trying with brew and it just keeps throwing other errors.

**Comment by stri8ed** [comment]
@xaksis I am on Linux (via Windows WSL), so used curl:

```
sudo curl -o /usr/local/bin/cog -L "https://github.com/replicate/cog/releases/v0.8.3/download/cog_$(uname -s)_$(uname -m)"
sudo chmod +x /usr/local/bin/cog
```

**Comment by peter65374** [comment]
Got same error as below 
Validating model schema...
```
ⅹ Model schema is invalid: invalid components: schema "Output": extra sibling fields: [title]
ⅹ
ⅹ {"components":{"schemas":{"Embedding":{"properties":{"embedding":{"items":{"type":"number"},"title":"Embedding","type":"array"},"shape":{"items":{"type":"integer"},"title":"Shape","type":"array"}},"required":["shape","embedding"],"title":"Embedding","type":"object"},"HTTPValidationError":{"properties":{"detail":{"items":{"$ref":"#/components/schemas/ValidationError"},"title":"Detail","type":"array"}},"title":"HTTPValidationError","type":"object"},"Input":{"properties":{"source_image":{"description":"input image file handler","format":"uri","title":"Source Image","type":"string","x-order":0}},"required":["source_image"],"title":"Input","type":"object"},"Output":{"$ref":"#/components/schemas/Embedding","title":"Output"},"PredictionRequest":{"properties":{"created_at":{"format":"date-time","title":"Created At","type":"string"},"id":{"title":"Id","type":"string"},"input":{"$ref":"#/components/schemas/Input"},"output_file_prefix":{"title":"Output File Prefix","type":"string"},"webhook":{"format":"uri","maxLength":65536,"minLength":1,"title":"Webhook","type":"string"},"webhook_events_filter":{"default":["start","output","logs","completed"],"items":{"$ref":"#/components/schemas/WebhookEvent"},"type":"array"}},"title":"PredictionRequest","type":"object"},"PredictionResponse":{"properties":{"completed_at":{"format":"date-time","title":"Completed At","type":"string"},"created_at":{"format":"date-time","title":"Created At","type":"string"},"error":{"title":"Error","type":"string"},"id":{"title":"Id","type":"string"},"input":{"$ref":"#/components/schemas/Input"},"logs":{"default":"","title":"Logs","type":"string"},"metrics":{"title":"Metrics","type":"object"},"output":{"$ref":"#/components/schemas/Output"},"started_at":{"format":"date-time","title":"Started At","type":"string"},"status":{"$ref":"#/components/schemas/Status"},"version":{"title":"Version","type":"string"}},"title":"PredictionResponse","type":"object"},"Status":{"description":"An enumeration.","enum":["starting","processing","succeeded","canceled","failed"],"title":"Status","type":"string"},"ValidationError":{"properties":{"loc":{"items":{"anyOf":[{"type":"string"},{"type":"integer"}]},"title":"Location","type":"array"},"msg":{"title":"Message","type":"string"},"type":{"title":"Error Type","type":"string"}},"required":["loc","msg","type"],"title":"ValidationError","type":"object"},"WebhookEvent":{"description":"An enumeration.","enum":["start","output","logs","completed"],"title":"WebhookEvent","type":"string"}}},"info":{"title":"Cog","version":"0.1.0"},"openapi":"3.0.2","paths":{"/":{"get":{"operationId":"root__get","responses":{"200":{"content":{"application/json":{"schema":{"title":"Response Root  Get"}}},"description":"Successful Response"}},"summary":"Root"}},"/health-check":{"get":{"operationId":"healthcheck_health_check_get","responses":{"200":{"content":{"application/json":{"schema":{"title":"Response Healthcheck Health Check Get"}}},"description":"Successful Response"}},"summary":"Healthcheck"}},"/predictions":{"post":{"description":"Run a single prediction on the model","operationId":"predict_predictions_post","parameters":[{"in":"header","name":"prefer","required":false,"schema":{"title":"Prefer","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/PredictionRequest"}}}},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/PredictionResponse"}}},"description":"Successful Response"},"422":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HTTPValidationError"}}},"description":"Validation Error"}},"summary":"Predict"}},"/predictions/{prediction_id}":{"put":{"description":"Run a single prediction on the model (idempotent creation).","operationId":"predict_idempotent_predictions__prediction_id__put","parameters":[{"in":"path","name":"prediction_id","required":true,"schema":{"title":"Prediction ID","type":"string"}},{"in":"header","name":"prefer","required":false,"schema":{"title":"Prefer","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"allOf":[{"$ref":"#/components/schemas/PredictionRequest"}],"title":"Prediction Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/PredictionResponse"}}},"description":"Successful Response"},"422":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HTTPValidationError"}}},"description":"Validation Error"}},"summary":"Predict Idempotent"}},"/predictions/{prediction_id}/cancel":{"post":{"description":"Cancel a running prediction","operationId":"cancel_predictions__prediction_id__cancel_post","parameters":[{"in":"path","name":"prediction_id","required":true,"schema":{"title":"Prediction ID","type":"string"}}],"responses":{"200":{"content":{"application/json":{"schema":{"title":"Response Cancel Predictions  Prediction Id  Cancel Post"}}},"description":"Successful Response"},"422":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HTTPValidationError"}}},"description":"Validation Error"}},"summary":"Cancel"}},"/shutdown":{"post":{"operationId":"start_shutdown_shutdown_post","responses":{"200":{"content":{"application/json":{"schema":{"title":"Response Start Shutdown Shutdown Post"}}},"description":"Successful Response"}},"summary":"Start Shutdown"}}}}
```
Does 0.8.6 cog fix it? Do I have to downgrade to 0.8.3?

**Comment by JimothyJohn** [comment]
I was also able to eliminate the error in 0.8.6 by not using the BaseModel in my output class so that seems to be the culprit. Instead I consolidated multiple Paths into a single zip file:

```py
# Started with this non-working version
class ModelOutput(BaseModel):
    prompt_npz: Optional[Path]
    audio_out: Path

class Predictor(BasePredictor):
    # setup code
    def predict(.....) -> ModelOutput

# Non-ideal but working version 
class Predictor(BasePredictor):
    # setup code
    def predict(.....) -> Path
```

**Comment by iamargentum** [comment]
had the same problem with -
```
class MattingOutput(BaseModel):
    success: bool
    error: Optional[str]
    segmentedImage: Optional[Path]
```

renaming `MattingOutput` to `Output` fixed it for me

**Comment by turian** [comment]
A gentler error message from cog/replicate would be useful, since this is an easy fix once you find this github issue.

**Comment by enesozturk** [comment]
> renaming MattingOutput to Output fixed it for me

Same here

**Comment by dkhokhlov** [comment]
Confirmed workaround:
> renaming MattingOutput to Output fixed it for me

looking into it

**Comment by dkhokhlov** [comment]
> Same here with bark model ( https://github.com/chenxwh/bark.git ). Any solutions? There's no just "Output" or "Matting".

@trogwarz 
The fix has been merged into mainline. The bark model build is ok:
```
bark$ cog build
....
 => => naming to docker.io/library/cog-bark                                                                                                                                               0.0s
Validating model schema...
Adding labels to image...
Unable to determine Git tag

Image built as cog-bark

$ cog --version
cog version v0.9.5+dev
```
This new cog v0.9.5 has not been released yet.  @nickstenning 

Meanwhile you could try to build cog yourself.

**Comment by dkhokhlov** [comment]
expected, v0.9.5 release was cut **before** my PR merge, on same day. Here is what was included in 0.9.5: 

https://github.com/replicate/cog/releases/tag/v0.9.5. 

so the fix will be in 0.9.6 cog release. note - local build from source still reports 0.9.5-dev. yes, confusing.

**Comment by dkhokhlov** [comment]
@trogwarz 0.9.6 released.
